src/extensions/timeseries/TimeWidgetNew/TimeSpanWidget.tsx

```diff
@@ -109,7 +111,14 @@ const createTimeSpanLayer = () => {
         const { position, anchor } = drag
         if (!position) return
         if (!anchor) return
-        const delta = position[0] - anchor[0]
+        let delta = position[0] - anchor[0]
+        if (!anchorTimeRange) return
+        if (anchorTimeRange.max + delta > numTimepoints) {
```

Comment:
During drag of time span handle. Tweak the behavior of the time span widget so that the current timepoint does not change relative to the time range when the time range is restricted from either left or right.